### ISSUE TITLE
Support for non-installed games

### ISSUE THREAD

**Issue by Solaire** [Nutzzz/GLC#49]
I have been porting the platform game detection to separate plugin projects (plus refactoring to improve overall code quality), and it got me thinking about support for non-installed games?

As far as I'm aware many platforms don't support installing games outside of their client (a quick google search for Steam got me nothing). Even if the big ones (Steam, Gog, Epic) offer some way, we can't guarantee that all will.

That brings me to my question. Should be support non-installed games? Given the above, I don't see the benefit of showing them to the user, since they can't really do anything with them. In version 2 of the app, uninstalled games will be kept in the database to preserve the statistics and the settings, but won't be shown in the app until the user installs them again.

**Comment by Nutzzz** [comment]
You're right that in some cases, attempting to start a non-installed game just launches the client and the user must then search for the game to install it, but for some platforms [including Steam, with caveat below], I've figured out a method to open the client directly to that game, and in at least one case it actually starts the install immediately.  I think (at least for those platforms), it certainly makes sense to optionally list those games.  There's a setting whether to show them or not in the general list or not, and we could certainly change the default to only show those games in the list of non-installed games.  In any case, I personally find the feature very helpful, if only to remember which platform it's on, and would love to extend this feature to the rest of the platforms (though some will require interfacing with web APIs).

EDIT: Here's the current state of things:
- Opens client and starts install: Epic, Steam*, Ubisoft
- Opens client directly to game: Amazon, GOG, itch
- Just opens client: Big Fish, Indiegala, Origin, Paradox
- Currently doesn't list non-installed games: Arc, Battle.net, Legacy, Oculus, Plarium, Riot, Rockstar, Wargaming

[EDIT 2022/8/19 to reflect PR #56 ]

'* = to show non-installed Steam games, game list must be set to public

**Comment by Solaire** [comment]
I see, thanks for the breakdown. I'll leave the non-installed games supported then and I'll update the platform API to support installation/removal (even if some clients won't support that at all)

**Comment by Nutzzz** [comment]
Note I just added a feature list to [my wiki](/Nutzzz/GLC/wiki/Features) which you might add to the official one, which includes a chart where features are platform-specific, including which platforms currently have non-installed game support.
[...though perhaps for the planned feature list you might link to a project page, or maybe just to the [issues with the "enhancement" label](https://github.com/Solaire/GLC/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement)].
